src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,168 @@
+(ns quo2.components.community.token-gating
+  (:require [quo.react-native :as rn]
+            [quo2.components.avatars.channel-avatar :as channel-avatar]
+            [quo2.components.buttons.button :as button]
+            [quo2.components.icon :as icon]
+            [quo2.components.markdown.text :as text]
+            [quo2.components.tags.token-tag :as token-tag]
+            [quo2.foundations.colors :as colors]
+            [quo.theme :as theme]
+            [status-im.i18n.i18n :as i18n]))
+
+(def styles {:container {:flex 1}
+             :header-container {:flex-direction :row
+
+                                :align-items :center}
+             :header-spacing-community {:margin-bottom 16}
+             :header-spacing-channel {:margin-bottom 24}
+             :header-avatar {:margin-right 8}
+             :header-community-avatar-image {:width 32 :height 32}
+             :header-title-container {:flex 1
+                                      :flex-direction :row
+                                      :align-items :center}
+             :header-title-lock {:margin-left 10}
+             :header-info-button {:margin-left 8}
+             :token-requirement-list-spacing {:padding-vertical 8}
+             :token-list {:align-items :center}
+             :token-row {:flex-direction :row}
+             :token-tag-spacing {:margin-left 7}
+             :divider {:height 1
+                       :margin-horizontal -20
+                       :margin-top 12
+                       :margin-bottom 8}
+             :enter-button {:margin-top 16}
+             :info-text {:margin-top 12
+                         :text-align :center
+                         :padding-horizontal 20}})
+
+(defn is-token-requirement-met? [token-requirement-list] (reduce
+                                                          #(or (and %1 (:is-sufficient %2)) false)
```

Comment:
1. Please move reduce to next line
2. can we use `(fn [var1 var2]` instead so it would be more clear what `%1` and `%2` are.
3. #(or ... false) ? (isn't or and false are redundant?

Reply:
1. ✅
2. sure!
3. If i understand correctly, `and` returns the value of what is passed to it, also if it's `nil` for example...

>  If a form returns logical false (nil or false), and returns that value and doesn't evaluate any of the other expressions, otherwise it returns the value of the last expr. (and) returns true.

I added `(or ... false)` so that `is-token-requirement-met?` function definitely always returns a boolean.

What do you think?

Reply:
I am not sure why you need boolean, nil is also a logical false. but anyway you can use something like this
```
(every?
 (fn [token]
   (get-in token [1 :is-sufficient]))
 token-requirement-list)
```

Reply:
Thanks a lot! I'm such a Clojure noob still, missing so many useful clojure functions 😅